### PR TITLE
modify tensorrt engine op from cpu mode to gpu 

### LINKED ISSUE #12719: refine tensorrt Integrate
1. Tensorrt engine op now just support cpu . All inputs of tensorrt engine op will be copied from gpu to cpu during inference process everytime.

2.  A pointer to CPU memory is needed of the TRT weight. 
  Before TRT runs, fluid loads weight into GPU storage. 
  so we need to copy the weights from GPU to CPU in our op converter. 
  If we use a temp tensor,  the weight memory will be released in advance, which affecting the construction of TRT Op.

### PR DESCRIPTION
**Pull request by NHZlX**
fix #12719

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/convert/conv2d_op.cc [hunk 1]
```diff
@@ -35,12 +35,20 @@ class Conv2dOpConverter : public OpConverter {
...
     PADDLE_ENFORCE_NOT_NULL(Y_v);
     auto* Y_t = Y_v->GetMutable<framework::LoDTensor>();
-    auto* weight_data = Y_t->mutable_data<float>(platform::CPUPlace());
 
-    PADDLE_ENFORCE_EQ(Y_t->dims().size(), 4UL);
-    const int n_output = Y_t->dims()[0];
-    const int filter_h = Y_t->dims()[2];
-    const int filter_w = Y_t->dims()[3];
+    platform::CPUPlace cpu_place;
+    framework::LoDTensor* weight_tensor = new framework::LoDTensor();
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/conv2d_op.cc]: wrap this raw pointer to a unique_ptr when it creates.

```
std::unique_ptr<framework::LoDTensor*> weight_tensor(new frameowrk::LoDTensor();
```

File: paddle/fluid/inference/tensorrt/convert/elementwise_op.cc [hunk 2]
```diff
@@ -40,10 +39,19 @@ class ElementwiseWeightOpConverter : public OpConverter {
     auto* Y_v = scope.FindVar(op_desc.Input("Y").front());
     PADDLE_ENFORCE_NOT_NULL(Y_v);
     auto* Y_t = Y_v->GetMutable<framework::LoDTensor>();
-    auto* weight_data = Y_t->mutable_data<float>(platform::CPUPlace());
+
+    platform::CPUPlace cpu_place;
+    framework::LoDTensor* weight_tensor = new framework::LoDTensor();
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/elementwise_op.cc]: same here

File: paddle/fluid/inference/tensorrt/convert/conv2d_op.cc (hunk 1 above)
Comment (NHZlX) [reply paddle/fluid/inference/tensorrt/convert/conv2d_op.cc]: I use the unique_ptr at  ```engine_->weight_map[op_desc.Input("Y").front()] =
        std::move(std::unique_ptr<framework::Tensor>(weight_tensor));```

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**